wagtail/admin/views/bulk_actions/move.py

```diff
@@ -0,0 +1,109 @@
+from urllib.parse import urlencode
+
+from django.core.exceptions import PermissionDenied
+from django.core.paginator import Paginator
+from django.shortcuts import get_list_or_404, get_object_or_404, redirect
+from django.template.response import TemplateResponse
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+from wagtail.admin import messages
+from wagtail.admin.views.pages.utils import get_valid_next_url_from_request
+from wagtail.core import hooks
+from wagtail.core.models import Page
+
+
+def move(request, parent_page_id, dest_page_id=None):
+    next_url = get_valid_next_url_from_request(request)
+    if not next_url:
+        next_url = reverse('wagtailadmin_explore', args=[parent_page_id])
+
+    page_ids = list(map(int, request.GET.getlist('id')))
+    child_pages = set()
+    if dest_page_id:
+        viewed_page = get_object_or_404(Page, id=dest_page_id)
+    else:
+        viewed_page = Page.get_first_root_node()
+
+    for _page in get_list_or_404(Page, id__in=page_ids):
+        page_to_move = _page.specific
+        page_perms = page_to_move.permissions_for_user(request.user)
+        if not page_perms.can_move():
```

Comment:
Can you select a page you don't have permissions to move? This could be an awkward user experience

Reply:
(as discussed in the meeting, these permissions/inability to move parts should probably be explicitly mentioned to the user on the confirmation view for each one)